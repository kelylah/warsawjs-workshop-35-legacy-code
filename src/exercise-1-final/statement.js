import createStatementData from './create-statement-data.js';

export default function statement(invoice, vehicles) {
  const statementData = createStatementData(invoice, vehicles);

  return renderPlainText(statementData, vehicles);
}

export function htmlStatement(invoice, vehicles) {
  const statementData = createStatementData(invoice, vehicles);

  return renderHtml(statementData, vehicles);
}

function renderPlainText(data) {
  let result = `Statement for ${data.customer}\n`;
  for (let ride of data.rides) {
    result += ` ${ride.vehicle.name}: ${pln(ride.amount)} (${
      ride.minutes
    } minutes)\n`;
  }
  result += `Amount owed is ${pln(data.totalAmount)}\n`;
  result += `You earned ${data.totalVolumeCredits} credits\n`;
  return result;
}

function renderHtml(data) {
  let result = `<h1>Statement for ${data.customer}</h1>\n`;
  result += '<table>\n';
  result += '<tr><th>vehicle</th><th>minutes</th><th>cost</th></tr>\n';
  for (let ride of data.rides) {
    result += `<tr><td>${ride.vehicle.name}</td><td>${ride.minutes}</td>`;
    result += `<td>${pln(ride.amount)}</td></tr>\n`;
  }
  result += '</table>\n';
  result += `<p>Amount owed is <em>${pln(data.totalAmount)}</em></p>\n`;
  result += `<p>You earned <em>${data.totalVolumeCredits}</em> credits</p>\n`;
  return result;
}

function pln(number) {
  return new Intl.NumberFormat('pl-PL', {
    style: 'currency',
    currency: 'PLN',
    minimumFractionDigits: 2,
  }).format(number / 100);
}
