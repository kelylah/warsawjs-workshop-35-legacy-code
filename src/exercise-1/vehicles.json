{
  "bike": { "name": "Bike", "type": "eco" },
  "scooter": { "name": "Scooter", "type": "electric" },
  "motorcycle": { "name": "Motorcycle", "type": "electric" }
}
