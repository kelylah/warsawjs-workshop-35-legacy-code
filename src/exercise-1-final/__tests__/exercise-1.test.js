import invoice from '../invoices.json';
import vehicles from '../vehicles.json';
import statement, { htmlStatement } from '../statement.js';

test('print statement in text', () => {
  const result = statement(invoice[0], vehicles);
  expect(result).toMatchSnapshot();
});

test('print statement in html', () => {
  const result = htmlStatement(invoice[0], vehicles);
  expect(result).toMatchSnapshot();
});
