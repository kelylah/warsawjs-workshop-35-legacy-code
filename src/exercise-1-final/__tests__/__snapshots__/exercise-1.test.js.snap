// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`print statement in html 1`] = `
"<h1>Statement for Michał</h1>
<table>
<tr><th>vehicle</th><th>minutes</th><th>cost</th></tr>
<tr><td>Bike</td><td>32</td><td>PLN 5.40</td></tr>
<tr><td>Scooter</td><td>10</td><td>PLN 15.00</td></tr>
<tr><td>Motorcycle</td><td>22</td><td>PLN 44.00</td></tr>
</table>
<p>Amount owed is <em>PLN 64.40</em></p>
<p>You earned <em>8</em> credits</p>
"
`;

exports[`print statement in text 1`] = `
"Statement for Michał
 Bike: PLN 5.40 (32 minutes)
 Scooter: PLN 15.00 (10 minutes)
 Motorcycle: PLN 44.00 (22 minutes)
Amount owed is PLN 64.40
You earned 8 credits
"
`;
